### PR TITLE
Refactors some reagent sacid/facid code

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Cleans up the code around facid/sacid/aceacid
Does some very minor balance changes/tweaks:
- Slightly increases sulphuric acid damage if you get splashed with more than 25 units

## Why It's Good For The Game
The code could use a cleanup, this makes it a lot more readable.

## Testing
I splashed a skrell with a LOT of acid

## Changelog
NPFC

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 1]
```diff
@@ -284,33 +284,37 @@
...
-						affecting.disfigure()
-				else
-					H.take_organ_damage(5, 10)
-			else
-				H.take_organ_damage(5, 10)
+	if(!ishuman(M) || isgrey(M))
+		return
+
+	var/mob/living/carbon/human/H = M
+	if(!(method == REAGENT_TOUCH))
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/toxins.dm:291]:
```suggestion
	if(method != REAGENT_TOUCH)
```

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 2]
```diff
@@ -284,33 +284,37 @@
...
-				H.emote("scream")
+			H.adjustFireLoss(clamp(volume, 1, 5))
+		return
+
+	if(H.wear_mask)
+		to_chat(H, "<span class='danger'>Your [H.wear_mask] protects you from the acid!</span>")
+		return
+	if(H.head)
+		to_chat(H, "<span class='danger'>Your [H.wear_mask] protects you from the acid!</span>")
+		return
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/toxins.dm:310]: I'm guessing this is what you meant to do here?
```suggestion
	if(H.head)
		to_chat(H, "<span class='danger'>Your [H.head] protects you from the acid!</span>")
		return
```

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 3]
```diff
@@ -383,26 +389,29 @@
...
-				H.adjustFireLoss(15)
-				H.emote("scream")
-			else
-				H.adjustBruteLoss(min(5, volume * 0.25))
-		else
-			to_chat(H, "<span class='warning'>The transparent acidic substance stings[volume < 25 ? " you, but isn't concentrated enough to harm you" : null]!</span>")
-			if(volume >= 25)
-				H.adjustBruteLoss(2)
-				H.emote("scream")
+	if(!ishuman(M))
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/toxins.dm:392]: You could probably just change the argument type here and avoid the istype cast

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 4]
```diff
@@ -383,26 +389,29 @@
...
+		return
+
+	var/mob/living/carbon/human/H = M
+	if(!(method == REAGENT_TOUCH))
```
Comment (lewcc) [inline code/modules/reagents/chemistry/reagents/toxins.dm:396]:
```suggestion
	if(method != REAGENT_TOUCH)
```

**Review by lewcc — COMMENTED** [review]
I really wish we had proper permeability, but I appreciate you cleaning up what we currently have.

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 2 above)
Comment (DGamerL) [reply code/modules/reagents/chemistry/reagents/toxins.dm:310]: Yep! Slight copy paste error

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 5]
```diff
@@ -337,42 +341,44 @@
...
+				to_chat(H, "<span class='danger'>Your [H.head.name] melts away!</span>")
+				qdel(H.head)
+			has_melted_something = TRUE
+
+		if(has_melted_something)
+			return
+
+	if(volume >= 5)
+		H.emote("scream")
+		H.adjustFireLoss(clamp((volume - 5) * 3, 8, 75))
```
Comment (DGamerL) [inline code/modules/reagents/chemistry/reagents/toxins.dm:380]:
```suggestion
		H.adjustFireLoss(clamp((volume - 5) * 3, 4, 75))
```
This should be lower in retrospect

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**